libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/KaptGenerateStubsTask.kt

```diff
@@ -16,65 +16,73 @@
 
 package org.jetbrains.kotlin.gradle.internal
 
-import org.gradle.api.artifacts.Configuration
+import org.gradle.api.file.ConfigurableFileCollection
 import org.gradle.api.file.DirectoryProperty
-import org.gradle.api.file.FileCollection
-import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
 import org.gradle.api.tasks.*
 import org.jetbrains.kotlin.cli.common.arguments.K2JVMCompilerArguments
-import org.jetbrains.kotlin.gradle.dsl.KotlinJvmOptions
 import org.jetbrains.kotlin.gradle.dsl.KotlinJvmOptionsImpl
+import org.jetbrains.kotlin.gradle.plugin.mpp.pm20.KotlinCompilationData
+import org.jetbrains.kotlin.gradle.tasks.AbstractKotlinCompile
 import org.jetbrains.kotlin.gradle.tasks.FilteringSourceRootsContainer
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 import org.jetbrains.kotlin.gradle.tasks.SourceRoots
-import org.jetbrains.kotlin.gradle.utils.getValue
 import org.jetbrains.kotlin.gradle.utils.isParentOf
 import org.jetbrains.kotlin.incremental.classpathAsList
 import org.jetbrains.kotlin.incremental.destinationAsFile
 import java.io.File
-import javax.inject.Inject
+import java.util.concurrent.Callable
 
 @CacheableTask
-open class KaptGenerateStubsTask @Inject constructor(
-    objectFactory: ObjectFactory
-) : KotlinCompile() {
+abstract class KaptGenerateStubsTask : KotlinCompile(KotlinJvmOptionsImpl()) {
+
+    class Configurator(private val kotlinCompileTask: KotlinCompile, kotlinCompilation: KotlinCompilationData<*>) :
+        AbstractKotlinCompile.Configurator<KaptGenerateStubsTask>(kotlinCompilation) {
+
+        override fun configure(task: KaptGenerateStubsTask) {
+            super.configure(task)
+            val providerFactory = kotlinCompileTask.project.providers
+            task.useModuleDetection.value(kotlinCompileTask.useModuleDetection).disallowChanges()
+            task.moduleName.value(kotlinCompileTask.moduleName).disallowChanges()
+            task.classpath = task.project.files(Callable { kotlinCompileTask.classpath })
+            task.kotlinTaskPluginClasspath.from(
+                providerFactory.provider { kotlinCompileTask.pluginClasspath }
```

Comment:
Nit: `providerFactory.provider {}` is not needed here, `ConfigurableFileCollection` could be populated lazily from another `FileCollection`